### PR TITLE
slcand: log to stdout when running in foreground

### PR DESCRIPTION
**Pull request by umlaeute**
when slcand runs in daemon mode, it makes sense to output via syslog.
but when running in foreground, it is nicer to have all the info on the stdout.

### CODE REVIEW

File: slcand.c [hunk 1]
```diff
@@ -54,6 +55,19 @@
...
+void fake_syslog(int priority, const char*format, ...)
+{
+	va_list ap;
+	printf("[%d] ", priority);
+	va_start(ap, format);
+	vprintf(format, ap);
+	va_end(ap);
+	printf("\n");
+}
+typedef void (*syslog_t)(int priority, const char*format, ...);
```
Comment (marckleinebudde) [inline slcand.c]: Please add a space before the *: `const char *format, ...);`

File: slcand.c [hunk 2]
```diff
@@ -54,6 +55,19 @@
...
+syslog_t syslogger;
+
+
```
Comment (marckleinebudde) [inline slcand.c]: one newline should be enough here

File: slcand.c [hunk 3]
```diff
@@ -54,6 +55,19 @@
 #define FLOW_HW 1
 #define FLOW_SW 2
 
+void fake_syslog(int priority, const char*format, ...)
+{
+	va_list ap;
```
Comment (marckleinebudde) [inline slcand.c]: one newline here, please

File: slcand.c [hunk 4]
```diff
@@ -185,6 +199,7 @@ int main(int argc, char *argv[])
 	int fd;
 
 	ttypath[0] = '\0';
+	syslogger = syslog;
```
Comment (marckleinebudde) [inline slcand.c]: you can move this to the declaration of `syslogger`

File: slcand.c [hunk 5]
```diff
@@ -54,6 +55,19 @@
...
+{
+	va_list ap;
+	printf("[%d] ", priority);
+	va_start(ap, format);
+	vprintf(format, ap);
+	va_end(ap);
+	printf("\n");
+}
+typedef void (*syslog_t)(int priority, const char*format, ...);
+syslog_t syslogger;
```
Comment (marckleinebudde) [inline slcand.c]: make it static please

**Comment by marckleinebudde** [comment]
Squashed and pushed to master. tnx.

**Comment by umlaeute** [comment]
thanks.
i *might* have caused less trouble if there was a documentation about `CONTRIBUTING` (detailing both the code style (<kbd>tab</kbd> indentention!) and the commit style (squash).

(though to be honest, it might as well not)

**Pull request closed without merging**